app/views/session.py

```diff
@@ -31,25 +33,32 @@ def login():
     :return: a 302 redirect to the next location for the user
     """
     logger.new()
-    # logging in again clears any session state
-    if session:
-        session.clear()
-
-    logger.debug("attempting token authentication")
-    authenticator.jwt_login(request)
-    logger.debug("token authenticated - linking to session")
-
-    metadata = get_metadata(current_user)
-
+    decrypted_token = decrypt_token(request.args.get('token'))
+    metadata = parse_metadata(decrypted_token)
     eq_id = metadata["eq_id"]
     form_type = metadata["form_type"]
-
-    logger.bind(eq_id=eq_id, form_type=form_type)
+    tx_id = metadata["tx_id"]
+    logger.bind(eq_id=eq_id, form_type=form_type, tx_id=tx_id)
 
     if not eq_id or not form_type:
```

Comment:
Could be but then need to break out the tx_id to bind before. Don't think it matters much so did it before